client/state/current-user/reducer.js

```diff
@@ -82,9 +90,28 @@ export const capabilities = createReducer( {}, {
 	}
 }, capabilitiesSchema );
 
+export const gravatarStatus = createReducer( {}, {
+	[ GRAVATAR_UPLOAD_START ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: true
+		} );
+	},
+	[ GRAVATAR_UPLOAD_SUCCESS ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: false
+		} );
+	},
+	[ GRAVATAR_UPLOAD_FAILURE ]: state => {
+		return Object.assign( {}, state, {
+			isUploading: false
+		} );
+	},
+}, gravatarSchema );
```

Comment:
I agree. So far, I was thinking of storing these three items in the state:
- date until which new Gravatar should be shown
- the Gravatar to be shown (local file ref or pulled from Gravatar by busting cache) 
- isUploading

I'd update `<Gravatar />` to show the new Gravatar if we're still within the date, and otherwise it'd fall back to the usual gravatar url (found under `currentUser.avatar_URL`).